### PR TITLE
[WEB-3841] 14.6.0 - Forms

### PR DESCRIPTION
**Pull request by jamiehenson**
## Jira Ticket Link / Motivation

More pricing page supporting stuff.

## Summary of changes

This PR is a mixture of feats, mixes and chores.

The main addition is the new `Forms` story page, which replaces the old MDX one and also houses the dropdowns and toggles which have been folded into it. New `input` stories showing dark theming have been added to this.

Also extra, a half-punt at dark mode in Storybook (it themes stories, doesn't seem to work with docs atm), and a fix for `Code` to format code lines with a right alignment. Chips have also been removed, as nothing uses them anywhere.

Next up, [this](https://ably.atlassian.net/browse/WEB-3907).

## Merge/Deploy Checklist

- [ ] Written automated tests for implemented features/fixed bugs
- [ ] Rebased and squashed commits
- [ ] Commits have clear descriptions of their changes
- [ ] Checked for any performance regressions

## Frontend Checklist

- [ ] No frontend changes in this PR
- [ ] Added before/after screenshots for changes
- [ ] Tested on different platforms/browsers with [Browserstack](https://www.browserstack.com)
- [ ] Compared with the initial design / our [brand guidelines](https://www.figma.com/file/jTQgyIovrhGBOf4Zrvjvbk/Ably-Linear-Design?node-id=118%3A0)
- [ ] Checked the code for accessibility issues ([VoiceOver User Guide](https://support.apple.com/en-gb/guide/voiceover/welcome/mac))?

### CODE REVIEW

File: .storybook/theme.ts [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
+
 export default create({
```
Comment (aralovelace) [inline .storybook/theme.ts:9]: My IDE is not happy with the `export default create` way lol. what if we use named export?

```
const theme = create({
  ...
});

export default theme; 
```

**Review by aralovelace — COMMENTED** [review]
Just few comments on the code that is not in this PR but in the file where the changes are. 

The new forms looks so good this is great! Question,  are the color for example toggle will always be always the same like the orange in the background?

**Review by kennethkalmer — COMMENTED** [review]
Looks good to me, however I'll defer final ✅ for @aralovelace in case I missed something

**Comment by kennethkalmer** [comment]
> The new forms looks so good this is great! Question, are the color for example toggle will always be always the same like the orange in the background?

I noticed this too actually

File: .storybook/preview.tsx [hunk 2]
```diff
@@ -3,7 +3,8 @@ import { DocsContainer } from "@storybook/blocks";
 import { initialize, mswLoader } from "msw-storybook-addon";
 
 import "./styles.css";
-import theme from "./theme";
+import { themes } from "@storybook/theming";
+import theme, { brandImage, brandImageDark } from "./theme";
 import loadIcons from "../src/core/icons";
 
 const docsContainer = ({ children, context, ...props }) => {
```
[the remark this answers is not in the document]
Comment (jamiehenson) [reply .storybook/preview.tsx:10]: Storybook's own typing for this appears to be `any`, `any` - so whilst this does seem sensible we don't really gain anything from adding this here

File: .storybook/theme.ts (hunk 1 above)
Comment (jamiehenson) [reply .storybook/theme.ts:9]: Sure, why not

**Comment by jamiehenson** [comment]
Good question with the checked toggle colour, you can achieve this with additional TW classes but this isn't portrayed in the Storybook example, I'll add that. In short, with the following
```ts
        <label className="ui-toggle">
          <input type="checkbox" className="peer" />
          <span className="ui-toggle-slider peer-checked:!bg-pink-600 bg-blue-800" />
        </label>
        <p>Custom color</p>
```
`ui-toggle-slider` can access a designated "peer" and modify its own styling state depending on what it does. This is nothing unordinary, it's Tailwind's way of doing sibling selectors.

**Review by kennethkalmer — APPROVED** [review]
I can't uncheck the pre-checked toggle for some reason, however I don't see this PR as the cause for that though (old code in the toggles story looks exactly the same as the new one)

**Review by aralovelace — APPROVED** [review]

**Pull request merged**